lib/iex/lib/iex/info.ex

```diff
@@ -3,12 +3,23 @@ defprotocol IEx.Info do
 
   @spec info(term) :: [{atom, String.t}]
   def info(term)
+
+  Kernel.def __protocols__(data) do
```

Comment:
@josevalim that's what I wanted to do but in there we don't have the information about the data type of the given `term`. We can do something like

```elixir
def i(term) do
  # What's the data type here?
  data_type =
    case term do
      term when is_bitstring(term) -> BitString
      ...
    end

  ...
end
```

but it doesn't look to good. We could reuse `Protocol.__builtin__/0` and such but I figured putting this in every implementation would be kinda straightforward. Am I missing something? :)